tests/serverpod_test_server/test_integration/serverpod_cli_args_override_test.dart

```diff
@@ -0,0 +1,342 @@
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
+
+    test(
+        'when applyRepairMigration is set to false in config and true in CLI args '
+        'then server applies repair migration', () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'test-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        ['--apply-repair-migration'],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      expect(serverpod.config.applyRepairMigration, isTrue);
+      // Other values should remain from config
+      expect(serverpod.config.runMode, equals('development'));
+      expect(serverpod.config.serverId, equals('test-server'));
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyMigrations, isFalse);
+    });
+
+    test(
+        'when multiple values are overridden by CLI args '
+        'then server uses CLI values for overridden fields and config values for others',
+        () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'config-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        [
+          '--mode',
+          'production',
+          '--server-id',
+          'cli-server',
+          '--apply-migrations'
+        ],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      // CLI overridden values
+      expect(serverpod.config.runMode, equals('production'));
+      expect(serverpod.config.serverId, equals('cli-server'));
+      expect(serverpod.config.applyMigrations, isTrue);
+      // Config values (not overridden)
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
+
+    test(
+        'when no CLI args are provided and only config is given '
+        'then server uses config values except runMode which defaults to development',
```

Comment:
Question: This sounds wrong? If I'm supplying the config with a runMode and I'm not overriding that with a commandline it should keep the runMode no?